src/main/scala/dpla/ingestion3/mappers/providers/MarylandMapping.scala

```diff
@@ -0,0 +1,125 @@
+import dpla.ingestion3.mappers.utils.{Document, XmlExtractor, XmlMapping}
+import dpla.ingestion3.model.DplaMapData._
+import dpla.ingestion3.model._
+import dpla.ingestion3.utils.Utils
+import org.json4s.JValue
+import org.json4s.JsonDSL._
+
+import scala.xml._
+
+class MarylandMapping extends XmlMapping with XmlExtractor {
+
+  // ID minting functions
+  override def useProviderName: Boolean = true
+
+  override def getProviderName: String = "maryland"
+
+  override def originalId(implicit data: Document[NodeSeq]): ZeroToOne[String] =
+    extractString(data \ "header" \ "identifier")
+
+  // SourceResource mapping
+
+  override def contributor(data: Document[NodeSeq]): ZeroToMany[EdmAgent] =
+    extractStrings(data \ "metadata" \\ "contributor")
+      .flatMap(_.splitAtDelimiter(";"))
+      .map(nameOnlyAgent)
+
+  override def creator(data: Document[NodeSeq]): ZeroToMany[EdmAgent] =
+    extractStrings(data \ "metadata" \\ "creator")
+      .flatMap(_.splitAtDelimiter(";"))
+      .map(nameOnlyAgent)
+
+  override def date(data: Document[NodeSeq]): ZeroToMany[EdmTimeSpan] =
+    extractStrings(data \ "metadata" \\ "date")
+      .flatMap(_.splitAtDelimiter(";"))
+      .map(stringOnlyTimeSpan)
+
+  override def description(data: Document[NodeSeq]): ZeroToMany[String] =
+    extractStrings(data \ "metadata" \\ "description")
+
+  override def format(data: Document[NodeSeq]): ZeroToMany[String] =
+    extractStrings(data \ "metadata" \\ "format")
+      .map(_.stripSuffix(";"))
+
+  override def language(data: Document[NodeSeq]): ZeroToMany[SkosConcept] =
+    extractStrings(data \ "metadata" \\ "language")
+      .flatMap(_.splitAtDelimiter(";"))
+      .map(nameOnlyConcept)
+
+  override def publisher(data: Document[NodeSeq]): ZeroToMany[EdmAgent] =
+    extractStrings(data \ "metadata" \\ "publisher")
+      .flatMap(_.splitAtDelimiter(";"))
+      .map(nameOnlyAgent)
+
+  override def subject(data: Document[NodeSeq]): ZeroToMany[SkosConcept] =
+    extractStrings(data \ "metadata" \\ "subject")
+      .flatMap(_.splitAtDelimiter(";"))
+      .map(nameOnlyConcept)
+
+  override def temporal(data: Document[NodeSeq]): ZeroToMany[EdmTimeSpan] =
+    extractStrings(data \ "metadata" \\ "temporal")
+      .flatMap(_.splitAtDelimiter(";"))
+      .map(stringOnlyTimeSpan)
+
+  override def title(data: Document[NodeSeq]): ZeroToMany[String] =
+    extractStrings(data \ "metadata" \\ "title")
+      .map(_.stripSuffix("."))
+
+  override def `type`(data: Document[NodeSeq]): ZeroToMany[String] =
+    extractStrings(data \ "metadata" \\ "type")
+      .flatMap(_.splitAtDelimiter(";"))
+
+  // OreAggregation
+  override def dplaUri(data: Document[NodeSeq]): ZeroToOne[URI] = mintDplaItemUri(data)
+
+  override def dataProvider(data: Document[NodeSeq]): ZeroToMany[EdmAgent] =
+    extractStrings(data \ "metadata" \\ "source")
+      .flatMap(_.splitAtDelimiter(";"))
+      .map(nameOnlyAgent)
+      .slice(0, 1) // get first instance
```

Comment:
Would `head` or `headOption` make more sense?

Reply:
This probably doesn't matter at all, just thinking out loud.

Reply:
Yeah, I just did it that way b/c `slice` returns an empty array if the value is missing, so you don't have to do another step to get the correct return type.

Reply:
TIL. That's awesome!